enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,825 @@
+To prevent both Machine API and Cluster API resources being reconciled,
+we will make sure that Cluster API controllers in the 4.N-1 release do not reconcile when a mirror resource exists with the same name.
+Independent of the authoritative API.
+
+This will mean that on downgrade all Machine management reverts to Machine API controllers.
+
+### Version Skew Strategy
+
+In 4.N-1, all resources that are mirrored into Cluster API will be managed by the Machine API controllers.
+The mirroring relationship will be based on the names of the resources to create a 1:1 mapping between Machines and MachineSets/ControlPlaneMachineSets in Machine API and Cluster API.
+If the Cluster API controller observes a mirror resource, it will not reconcile, independent of the value of the authoritative annotation.
+
+During upgrade, we expect all Machines to be authoritative in Machine API, however, if resources are marked as authoritative in Cluster API,
+there may be a small amount of time[^10] during upgrade where the resources are not reconciled, between the time that the Machine API controllers are upgraded and the Cluster API controllers are upgraded.
+
+[^10]: We must ensure that Cluster API controllers are upgraded _after_ Machine API controllers to ensure that this upgrade is safe.
+The Machine API controllers must observe the annotation and stop reconciling before the Cluster API controllers observe the annotation and act upon the value.
+This can be done by running the Cluster API resources at run-level 31 for the release in which this upgrade is handled.
+
+### Operational Aspects of API Extensions
+
+The new synchronisation controller will operator in the `openshift-machine-api` namespace alongside existing Machine API controllers.
+It forms the bases of the conversion between the two API groups, but as a controller, rather than a conversion webhook.
+Its SLIs and impact on the cluster are therefore different to a traditional conversion webhook.
+
+We expect, when the sync controller is failing, that it will report individual errors as conditions on the Machine API resources.
+For wider failures, it will report a condition on the `machine-api` ClusterOperator, which will be aggregated with other conditions to determine the health of the Machine API overall.
+
+There should be no impact on the operation of existing systems by the new synchronisation logic; it does not affect the functionality, rather it mirrors the resource for informational purposes primarily.
+
+#### Failure Modes
+
+When non-functional, the sync controller will not update the non-authoritative resources in the cluster.
+This could lead to stale information being present in the non-authoritative API.
+
+This could lead to a risk that the authoritative API is switched while the data is stale.
+
+Since the sync controller is required to update the `status.authoritativeAPI` to enact the switch, the desired switch will not happen until the sync controller is restored.
```

Comment:
I meant, if the sync controller is broken, then the switch to the authoritative APi won't actually happen until the sync controller starts working again.

So the failure mode here is that while the user requested an action, the action is not actioned because the controller is non-functional, but, that's fine, because apart from an alert, nothing will break